source/h1_connection.c

```diff
@@ -418,6 +418,42 @@ static void s_connection_update_window(struct aws_http_connection *connection_ba
     }
 }
 
+int aws_h1_stream_activate(struct aws_http_stream *stream) {
+    struct aws_h1_stream *h1_stream = AWS_CONTAINER_OF(stream, struct aws_h1_stream, base);
+
+    struct aws_http_connection *base_connection = stream->owning_connection;
+    struct h1_connection *connection = AWS_CONTAINER_OF(base_connection, struct h1_connection, base);
+
+    bool should_schedule_task = false;
+
+    { /* BEGIN CRITICAL SECTION */
+        s_h1_connection_lock_synced_data(connection);
+        stream->id = aws_http_connection_get_next_stream_id(base_connection);
```

Comment:
Check to make sure stream->id isn't already set.
Yes, it's a programmer error, but `.activate()` is going to be public all the way out the the CRTs and they're often not compiled with our asserts turned on

Reply:
I guess this could just no-op if it's already activated. Ain't no harm done